charms/admission-webhook/reactive/admission.py

```diff
@@ -0,0 +1,142 @@
+import os
+from base64 import b64encode
+from pathlib import Path
+from subprocess import run
+
+import yaml
+
+from charmhelpers.core import hookenv
+from charms import layer
+from charms.reactive import clear_flag, hook, set_flag, when, when_not
+
+
+@hook('upgrade-charm')
+def upgrade_charm():
+    clear_flag('charm.started')
+
+
+@when('charm.started')
+def charm_ready():
+    layer.status.active('')
+
+
+@when('layer.docker-resource.oci-image.changed')
+def update_image():
+    clear_flag('charm.started')
+
+
+@when('layer.docker-resource.oci-image.available')
+@when_not('charm.started')
+def start_charm():
+    layer.status.maintenance('configuring container')
+
+    image_info = layer.docker_resource.get_info('oci-image')
+    model = os.environ['JUJU_MODEL_NAME']
+
+    run(
+        [
+            "openssl",
+            "req",
+            "-x509",
+            "-newkey",
+            "rsa:4096",
+            "-keyout",
+            "key.pem",
+            "-out",
+            "cert.pem",
+            "-days",
+            "365",
+            "-subj",
+            f"/CN={hookenv.service_name()}.{model}.svc",
+            "-nodes",
```

Comment:
Ok, you could put *some* of these on the same lines to shorten this, lol

Reply:
It's formatted using black, which has some formatting choices I don't like but overall not enough to stop using it.